yt/frontends/gizmo/data_structures.py

```diff
@@ -49,3 +54,113 @@ def _is_valid(cls, filename, *args, **kwargs):
 
     def _set_code_unit_attributes(self):
         super()._set_code_unit_attributes()
+
+    def _parse_parameter_file(self):
+        hvals = self._get_hvals()
+
+        self.dimensionality = 3
+        self.refine_by = 2
+        self.parameters["HydroMethod"] = "sph"
+        # Set standard values
+
+        # We may have an overridden bounding box.
+        if self.domain_left_edge is None and hvals["BoxSize"] != 0:
+            self.domain_left_edge = np.zeros(3, "float64")
+            self.domain_right_edge = np.ones(3, "float64") * hvals["BoxSize"]
+
+        self.domain_dimensions = np.ones(3, "int32")
+        self._periodicity = (True, True, True)
+
+        self.cosmological_simulation = 1
+
+        try:
+            self.current_redshift = hvals["Redshift"]
+        except KeyError:
+            # Probably not a cosmological dataset, we should just set
+            # z = 0 and let the user know
+            self.current_redshift = 0.0
+            only_on_root(mylog.info, "Redshift is not set in Header. Assuming z=0.")
+
+        try:
+            # The current version of GIZMO has updated the names of the Omegas
+            # to include an _, added baryons and radiation and added the
+            # ComovingIntegrationOn field. ComovingIntegrationOn is always set,
+            # but the Omega's are only included if ComovingIntegrationOn is true
+            if "ComovingIntegrationOn" in hvals:
+                self.cosmological_simulation = hvals["ComovingIntegrationOn"]
+                self.omega_lambda = hvals["Omega_Lambda"]
+                self.omega_matter = hvals["Omega_Matter"]
+                self.omega_baryon = hvals["Omega_Baryon"]
+                self.omega_radiation = hvals["Omega_Radiation"]
+                self.hubble_constant = hvals["HubbleParam"]
+            else:
+                # Should still support old GIZMO versions too
+                only_on_root(
+                    mylog.info,
+                    "ComovingIntegrationOn does not exist, falling back to OmegaLambda",
+                )
+                self.omega_lambda = hvals["OmegaLambda"]
+                self.omega_matter = hvals["Omega0"]
+                self.hubble_constant = hvals["HubbleParam"]
+                self.cosmological_simulation = self.omega_lambda == 0.0
+        except KeyError:
+            # If these are not set it is definitely not a cosmological dataset.
+            self.omega_lambda = 0.0
+            self.omega_matter = 1.0  # Just in case somebody asks for it.
```

Comment:
non-cosmological datasets normally have `omega_matter` set to `0.0`. Is there a reason not to follow the convention here ?